docs/topics/security.txt

```diff
@@ -172,18 +210,98 @@ vulnerability allows and much worse).
+CSRF token pairs (cookie and POST data) are:
+
+* arbitrary
+* not linked to sessions by default.
+
+Arbitrary CSRF token pairs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This will occasionally be flagged in security audits as a vulnerability. That
+is, when Django checks the cookie against the POSTed hidden field, it only
+checks that they match *each other*, not that they match the server secret on
+which they are both based, or that they can be verified against the user's
+session.
+
+This means an attacker could assemble a POST request on a victim's machine,
+containing a matching cookie-field pair, that would pass the CSRF checks.
+
+It's possible to stage a 'proof' of such an attack by manipulating one's own
+cookies and constructing malicious POST requests that defeat CSRF protection.
+This would be akin to complaining of a weakness in one's own front-door by
+demonstrating that it can be opened by someone inside the house - in each case,
+an attack of this kind is predicated on a high level of access already.
+
+In a real CSRF attack, the attacker would need access to the victim's cookies
+to be able to achieve this, via (for example) a man-in-the-middle or cross-site
+scripting attack, in which case the victim has already been thoroughly
+compromised, and exploiting CSRF is almost certainly an unnecessary extra
+effort.
+
+CSRF tokens not associated with sessions by default
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CSRF tokens *can* be linked to sessions by using the
+:setting:`CSRF_USE_SESSIONS`, which stores them in the user's session cookie.
+(note that this would provide no additional protection against the
+CSRF-defeating attacks described above).
+
+However this is not the default, because *not* linking CSRF protection to a
+session allows using the protection on sites (for example, pastebins such as
+`dpaste.com <http://dpaste.com>`_ that allow submissions from anonymous users
+who don't have a session.
+
+Rationale for Django's CSRF protection design decisions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Django's CSRF protection is realistic; it doesn't attempt to defend against
+  vulnerabilities that are premised upon much more serious compromises.
+* Django's defaults are stateless (that is, they don't depend on server state).
+  This allows it to scale well, for example in deployments based on multiple
+  application server instances.
+* Deployments at scale will often use backends that don't require synchronized
+  locking access to server side data.
+* Not tying CSRF protection to sessions allows developers to use different
+  session backends, without the risk of relying on a backend that may not
+  have implemented CSRF protection correctly.
+* Django's technique supports forms in multiple browser tabs, and not just for
+  the last form loaded.
+* Django doesn't oblige the site to create anonymous sessions to support forms,
+  helping eliminate a Denial of Service vulnerability (creating many anonymous
```

Comment:
adding `DoS` here may help. Adding the short explanation after is good though, in lieu of any other internal docs on the topic